### PR TITLE
[WIP] Broken Website Link PR Check

### PR DESCRIPTION
**Pull request by jescalan**
A github action that runs against PRs that change the website, checking to see if any links or anchor links are broken by the changes.

### CODE REVIEW

File: .github/actions/check_links/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+async function run() {
+  const id = await createCheck();
+  const root = path.join(__dirname, "../../..");
+  const deployUrl = core.getInput("baseUrl", { required: true });
+
+  try {
+    // Run the link check against the PR preview link
+    const output = execSync(`./node_modules/.bin/linkcheck ${deployUrl}`, {
```
Comment (pruett) [inline .github/actions/check_links/index.js]: Alternative package/tool: https://github.com/actions/toolkit/tree/master/packages/exec

File: .github/actions/check_links/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      cwd: root
+    });
+
+    console.log(output);
+    await updateCheck(id, { output: { summary: output } });
```
Comment (pruett) [inline .github/actions/check_links/index.js]: **Note**: `output` will need to conform to GitHub checks API

File: .github/actions/workflows/check_links.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: HashiCorp PR Bot
+on: [deployment_status]
+jobs:
+  check_links:
+    if: github.event.deployment.environment == 'Netlify Preview' && github.event.deployment_status.state == 'success' && github.event.deployment.payload.context == 'deploy-preview'
+    name: Check links
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - run: npm install
```
Comment (pruett) [inline .github/actions/workflows/check_links.yml]:
```yaml
- run: npm install
  working-directory: ./website
```

File: .github/actions/workflows/check_links.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      - uses: ./.github/actions/check-links
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          baseUrl: ${{ github.event.deployment.payload.environment_url }}
```
Comment (pruett) [inline .github/actions/workflows/check_links.yml]: looks like you'll need to declare `working-directory` for every step

```yaml
      - uses: ./.github/actions/check-links
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          baseUrl: ${{ github.event.deployment.payload.environment_url }}
        working-directory: ./website     
```

File: .github/actions/workflows/check_links.yml [hunk 5]
```diff
@@ -8,6 +8,7 @@ jobs:
     steps:
       - uses: actions/checkout@master
       - run: npm install
+      - working-directory: ./website
```
Comment (pruett) [inline .github/actions/workflows/check_links.yml]:
```suggestion
      - run: npm install
        working-directory: ./website
```

Comment (pruett) [reply .github/actions/workflows/check_links.yml]: I think it's a nested property

File: .github/actions/workflows/check_links.yml [hunk 6]
```diff
@@ -8,7 +8,7 @@ jobs:
       - uses: actions/checkout@master
       - run: npm install
         working-directory: ./website
-      - uses: ./.github/actions/check-links
+      - uses: ./.github/actions/check_links
```
Comment (pruett) [inline .github/actions/workflows/check_links.yml]: 😍 Makes more sense now!

File: .github/actions/check_links/index.js [hunk 7]
```diff
@@ -38,9 +38,10 @@ async function run() {
 
   try {
     // Run the link check against the PR preview link
-    const output = execSync(
-      `../../../website/node_modules/.bin/linkcheck ${deployUrl}`
-    );
+    // const output = execSync(
+    //   `../../../website/node_modules/.bin/linkcheck ${deployUrl}`
+    // );
+    const output = execSync(`ls ../../../website/node_modules/.bin`);
```
Comment (pruett) [inline .github/actions/check_links/index.js]: for path related stuff i'd check out `GITHUB_WORKSPACE`.

Try logging `process.env.GITHUB_WORKSPACE` and working from there.

Comment (pruett) [reply .github/actions/check_links/index.js]: https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables

**Comment by jescalan** [comment]
Closing as this is now stale - a bug in github actions that github support was unable to resolve is preventing us from being able to merge this. Planning to pivot into a github app instead of actions to allow this to continue moving forward. Work on this is slated for next week.

**Pull request closed without merging**